WasmEdge/WasmEdge
# WASI Filesystem refactor

* Filesystem refactor
    * Use VINode for path resolve and capability check.
    * Use INode for system-dependended file actions.
    * Use Environ for file descriptor holding, other miscellaneous actions.
    * Add Clock for system-dependended clock actions.
    * Memory check and valid flags check are in host functions.
    * Add kqueue for MacOS X
    * Support MacOS X
    * Partially support Windows 10
* Other
    * Add wasi-test for testing basic WASI interface
    * Disable blake3 AVX512 support on old platform
    * Ensure every platform defines is defined
